### PR TITLE
Support for fieldcollection queries

### LINKED ISSUE #53: Support Fieldcollection queries

### PR DESCRIPTION
**Pull request by weisswurstkanone**
resolves #53

### CODE REVIEW

File: src/GraphQL/Type/FieldcollectionType.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function resolveType($element, $context, ResolveInfo $info)
+    {
+        if ($element instanceof FieldcollectionDecriptor) {
+
+            $eDump = $element->getArrayCopy();
```
Comment (NiklasBr) [inline src/GraphQL/Type/FieldcollectionType.php]: Used anywhere?

File: src/GraphQL/Type/PimcoreObjectType.php [hunk 2]
```diff
@@ -109,21 +121,125 @@ public function build($context = [])
...
+            } else {
+                $fcDef = Definition::getByKey($allowedFcName);
+                $fcFields = [];
+
+
+                $fcFieldDefs = $fcDef->getFieldDefinitions();
+
+                foreach ($fcFieldDefs as $key => $fieldDef) {
+                    $attrName = $fieldDef->getName();
+                    $columnDesc = array(
```
Comment (NiklasBr) [inline src/GraphQL/Type/PimcoreObjectType.php]: Aesthetics only, but you are mixing short and long array syntax :)

File: src/GraphQL/Type/PimcoreObjectType.php [hunk 3]
```diff
@@ -95,11 +101,17 @@ public function build($context = [])
...
-                    if ($result) {
-                        $fields[$result['key']] = $result['config'];
+
+                    if (!$column["isOperator"] && is_array($column["attributes"]) && $column["attributes"]["dataType"] == "fieldcollections") {
+                        $this->addFieldCollectionDefs($column, $class, $fields);
+
+                    } else {
+                        /** @var $fieldHelper DataObjectFieldHelper */
+                        $fieldHelper = $this->getGraphQlService()->getObjectFieldHelper();
+                        $result = $fieldHelper->getQueryFieldConfigFromConfig($column, $class);
```
Comment (NiklasBr) [inline src/GraphQL/Type/PimcoreObjectType.php]: `getQueryFieldConfigFromConfig` can return `bool|array`, what happens with `$fields[$result['key']]` if `$result` is `true` will it throw?

File: src/GraphQL/Type/FieldcollectionType.php (hunk 1 above)
Comment (weisswurstkanone) [reply src/GraphQL/Type/FieldcollectionType.php]: thx!

**Pull request merged**